sources/api/migration/migrator/src/main.rs

```diff
@@ -349,14 +237,38 @@ where
     let mut intermediate_datastores = HashSet::new();
 
     for migration in migrations {
+        // get the migration from the repo
+        let lz4_bytes = repository
+            .read_target(migration.as_str())
+            .context(error::LoadMigration {
+                migration: migration.to_owned(),
+            })?
+            .context(error::MigrationNotFound {
+                migration: migration.to_owned(),
+            })?;
+
+        // deflate with an lz4 decoder read
+        let mut reader = lz4::Decoder::new(lz4_bytes).context(error::Lz4Decode { migration })?;
+
+        // TODO - remove this use of the filesystem when we add pentacle
+        let exec_path = migrations_rundir.join(&migration);
+        {
+            let mut f = OpenOptions::new()
+                .write(true)
+                .create(true)
+                .open(&exec_path)
+                .context(error::MigrationSave {
+                    path: exec_path.clone(),
```

Comment:
```suggestion
                    path: &exec_path,
```
Should be able to let Snafu handle that, and only clone on error cases.